### ISSUE TITLE
Error when running the gos:websocket:server

### ISSUE THREAD

**Issue by parhamdoustdar** [GeniusesOfSymfony/WebSocketBundle#34]
I've registered the bundle in my Symfony application, and this is what I get when I try and run the server:

```

Fatal error: Declaration of Gos\Bundle\WebSocketBundle\Router\NullPubSubRouter::generate() must be compatible with Gos\Bundle\PubSubRouterBundle\Router\RouterInterface::generate($routeName, array $parameters = Array, $tokenSeparator = NULL) in /vagrant/code/vendor/gos/web-socket-bundle/Router/NullPubSubRouter.php on line 10

Call Stack:
    0.0013     232760   1. {main}() /vagrant/code/app/console:0
    0.2934    3508528   2. Symfony\Component\Console\Application->run() /vagrant/code/app/console:27
    0.3376    3710992   3. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() /vagrant/code/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:126
    1.8473    6149040   4. Symfony\Bundle\FrameworkBundle\Console\Application->registerCommands() /vagrant/code/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:73
    4.3955   18654640   5. Symfony\Component\DependencyInjection\Container->get() /vagrant/code/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:111
    4.3955   18654960   6. appDevDebugProjectContainer->getGosWebSocket_ServerCommandService() /vagrant/code/app/bootstrap.php.cache:2140
    4.4054   18709560   7. Symfony\Component\DependencyInjection\Container->get() /vagrant/code/app/cache/dev/appDevDebugProjectContainer.php:2412
    4.4054   18709848   8. appDevDebugProjectContainer->getGosWebSocket_EntryPointService() /vagrant/code/app/bootstrap.php.cache:2140
    4.4104   18719736   9. Symfony\Component\DependencyInjection\Container->get() /vagrant/code/app/cache/dev/appDevDebugProjectContainer.php:2231
    4.4104   18720040  10. appDevDebugProjectContainer->getGosWebSocket_Server_RegistryService() /vagrant/code/app/bootstrap.php.cache:2140
    4.4171   18726680  11. Symfony\Component\DependencyInjection\Container->get() /vagrant/code/app/cache/dev/appDevDebugProjectContainer.php:2397
    4.4171   18727064  12. appDevDebugProjectContainer->getGosWebSocket_Ws_ServerService() /vagrant/code/app/bootstrap.php.cache:2140
    4.4605   18857208  13. Symfony\Component\DependencyInjection\Container->get() /vagrant/code/app/cache/dev/appDevDebugProjectContainer.php:6565
    4.4605   18857592  14. appDevDebugProjectContainer->getGosWebSocketServer_WampApplicationService() /vagrant/code/app/bootstrap.php.cache:2140
    4.5006   18947704  15. Symfony\Component\DependencyInjection\Container->get() /vagrant/code/app/cache/dev/appDevDebugProjectContainer.php:6584
    4.5006   18948056  16. appDevDebugProjectContainer->getGosWebSocket_Topic_DispatcherService() /vagrant/code/app/bootstrap.php.cache:2140
    4.5204   18996704  17. Symfony\Component\DependencyInjection\Container->get() /vagrant/code/app/cache/dev/appDevDebugProjectContainer.php:2427
    4.5204   18997024  18. appDevDebugProjectContainer->getGosWebSocket_Router_WampService() /vagrant/code/app/bootstrap.php.cache:2140
    4.5248   19015816  19. Symfony\Component\DependencyInjection\Container->get() /vagrant/code/app/cache/dev/appDevDebugProjectContainer.php:2330
    4.5248   19016088  20. appDevDebugProjectContainer->getGosWebSocket_Null_Pubsub_RouterService() /vagrant/code/app/bootstrap.php.cache:2140
    4.5249   19016376  21. spl_autoload_call() /vagrant/code/app/bootstrap.php.cache:2259
    4.5249   19016448  22. Symfony\Component\Debug\DebugClassLoader->loadClass() /vagrant/code/app/bootstrap.php.cache:2259
    4.5291   19029664  23. require('/vagrant/code/vendor/gos/web-socket-bundle/Router/NullPubSubRouter.php') /vagrant/code/vendor/symfony/symfony/src/Symfony/Component/Debug/DebugClassLoader.php:153

```

**Comment by ProPheT777** [comment]
Fixed in v1.3.1, thank to reporting it.

**Comment by parhamdoustdar** [comment]
I manually changed this to add the `null` value so that the two function signatures would be the same. However, I then got another error about the `generateWithTokens()` not matching the `GeneratorInterface`.

Also, even the `generatorInterface::generateWithTokens()` has an issue. The last argument is not optional, but the argument before it _is_ optional:

```
public function generateFromTokens(RouteInterface $route, Array $tokens, Array $parameters = [], $tokenSeparator);
```

`$tokenSeparator` must also have a default value, or it must be moved to not be the last argument.
